src/gui/dbsettings/DatabaseSettingsWidgetStatistics.cpp

```diff
@@ -0,0 +1,207 @@
+
+/*
+ *  Copyright (C) 2019 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "DatabaseSettingsWidgetStatistics.h"
+#include "ui_DatabaseSettingsWidgetStatistics.h"
+
+#include "core/Database.h"
+#include "core/FilePath.h"
+#include "core/Group.h"
+#include "core/Metadata.h"
+
+#include <QFileInfo>
+#include <QMessageBox>
+#include <QStandardItemModel>
+#include <map>
+
+namespace
+{
```

Comment:
You don't need a namespace, the class encapsulates everything on its own.

Reply:
I once had a program crash (in Windows, built with VC++, don't remember the version) because a class with identical name but different contents was defined in another cpp file, both globally visible (not in an anonymous namespace). I can change it if you wish, but if someone defines a "class Stats" somewhere else on global level, Windows people might run into problems.

Reply:
You are most likely correct. We can keep it this way, it doesn't harm anything.